homeassistant/components/climate/mysensors.py

```diff
@@ -0,0 +1,238 @@
+        """Return true if away mode is on."""
+        return self._aux
+
+    @property
+    def current_fan_mode(self):
+        """Return the fan setting."""
+        return self._current_fan_mode
+
+    @property
+    def fan_list(self):
+        """List of available fan modes."""
+        return self._fan_list
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        set_req = self.gateway.const.SetReq
+        if kwargs.get(ATTR_TEMPERATURE) is not None:
+            self._target_temperature = kwargs.get(ATTR_TEMPERATURE)
+        self.gateway.set_child_value(self.node_id, self.child_id,
+                                     set_req.V_HVAC_SETPOINT_COOL,
+                                     self._target_temperature)
+        if self.gateway.optimistic:
+            # optimistically assume that device has changed state
+            self._values[set_req.V_HVAC_SETPOINT_COOL] = \
+                            self._target_temperature
+        self.update_ha_state()
+
+    def set_humidity(self, humidity):
+        """Set new target temperature."""
+        self._target_humidity = humidity
+        self.update_ha_state()
+
+    def set_swing_mode(self, swing_mode):
+        """Set new target temperature."""
+        self._current_swing_mode = swing_mode
+        self.update_ha_state()
+
+    def set_fan_mode(self, fan):
+        """Set new target temperature."""
+        set_req = self.gateway.const.SetReq
+        self._current_fan_mode = fan
+        self.gateway.set_child_value(self.node_id, self.child_id,
+                                     set_req.V_HVAC_SPEED, fan)
+        if self.gateway.optimistic:
+            # optimistically assume that switch has changed state
+            self._values[set_req.V_HVAC_SPEED] = fan
+        self.update_ha_state()
+
+    def set_operation_mode(self, operation_mode):
+        """Set new target temperature."""
+        set_req = self.gateway.const.SetReq
+        self._current_operation = operation_mode
+        self.gateway.set_child_value(self.node_id, self.child_id,
+                                     set_req.V_HVAC_FLOW_STATE, operation_mode)
+        if self.gateway.optimistic:
+            # optimistically assume that switch has changed state
+            self._values[set_req.V_HVAC_FLOW_STATE] = operation_mode
+        self.update_ha_state()
+
+    @property
+    def current_swing_mode(self):
+        """Return the swing setting."""
+        return self._current_swing_mode
+
+    @property
+    def swing_list(self):
+        """List of available swing modes."""
+        return self._swing_list
+
+    def turn_away_mode_on(self):
+        """Turn away mode on."""
+        self._away = True
+        self.update_ha_state()
+
+    def turn_away_mode_off(self):
+        """Turn away mode off."""
+        self._away = False
+        self.update_ha_state()
+
+    def turn_aux_heat_on(self):
+        """Turn away auxillary heater on."""
+        self._aux = True
```

Comment:
See above. Mind that it will be hard to support a way of communicating this directly to mysensors as support is lacking there. You could keep it to leave option open for indirect change via another entity.